packages/react/dom.js

```diff
@@ -44,7 +44,11 @@ exports.render = function (reactElement, context) {
     return context.bootstrap().then(function () {
       var enhancedElement = context.enhanceElement(reactElement);
       return context.prepareRender(enhancedElement).then(function () {
-        ReactDOM.render(enhancedElement, mountpoint);
+        if (ReactDOM.hydrate) {
```

Comment:
Is it ok to call `.hydrate()` if there's no server-side rendered markup?
I assume it would just work because then there's a tree mismatch at the root level and everything will have to be rerendered. Does it print any warnings or is there maybe a performance penalty for doing this instead of using `.render()` in a purely client-side application?

Reply:
I think we can always expect *some* sort of server side rendering to have occurred in a Hops context - either at build- or runtime. Accordingly, we should be safe to always `hydrate`, shouldn't we..?

Reply:
Btw. I just verified that hydrating without pre-rendered markup apparently works, but according to the [ReactDOM docs](https://reactjs.org/docs/react-dom.html#hydrate), "you should treat mismatches as bugs and fix them".

Once again, in a hops-react context, I would always expect server side rendering to occur. The above-mentioned docs contain some hints on how to deal with necessary differences in server/browser rendering.